frontend/src/components/mainPage/codeEditor/CodeEditor.vue

```diff
@@ -94,7 +94,7 @@
                     }
 
                     let legacyChange = monacoChangeToLegacyFormat(this.$props.code, change);
-                    moveTrackedVariables(legacyChange);
+                    moveTrackedVariables(this.project, legacyChange);
```

Comment:
Jak już to potrzebujemy tam tylko code, więc nie przekazywałbym całego projektu tylko code.

Reply:
Nie mierzyłem tego, ale zakładam, że przekazanie referencji do `project` jest szybsze niż kopiowanie całego `code` (jeśli się nie mylę `code` to zwykły string, więc zostanie przekopiowany). Dodatkowo gdzieś tam wgłąb tej funkcji jeszcze raz przekazujemy ten projekt czy kod, więc dla każdej zmiennej cały kod musiałby być kopiowany w pamięci. Być może nasze narzędzia by to zoptymalizowały, ale wolałem z góry założyć, że nie chcę kopiować całego stringa.